### PR TITLE
chore(mappings): migrate legacy mappings under the hood

### PR DESCRIPTION
**Pull request by kyazdani42**
NOTE: this should not be merged before we tested it thoroughly.
NOTE2: this should not be merged before the help menu is rewritten
NOTE3: this can be used to print the on_attach migration to the user.

### CODE REVIEW

File: lua/nvim-tree.lua [hunk 1]
```diff
@@ -666,8 +666,9 @@ function M.setup(conf)
   validate_options(conf)
 
   local opts = merge_options(conf)
-  local netrw_disabled = opts.disable_netrw or opts.hijack_netrw
+  legacy.move_mappings_to_keymap(opts)
```
Comment (alex-courtis) [inline lua/nvim-tree.lua:669]: I like this approach: moving the mappings after merge means we only have to do this once.

File: lua/nvim-tree/legacy.lua [hunk 2]
```diff
@@ -297,6 +298,99 @@ local function removed(opts)
   end
 end
 
+local OLD_MAPPING_TABLE = {
```
Comment (alex-courtis) [inline lua/nvim-tree/legacy.lua:301]: We could be lazy and move this to `DEFAULT_KEYMAPS` with a `legacy_action` field.

**Review by alex-courtis — COMMENTED** [review]
> NOTE2: this should not be merged before the help menu is rewritten

We could be lazy: retain the view.mappings.list for help to look up, if we want to merge this before.

> NOTE3: this can be used to print the on_attach migration to the user.

Yes. `vim.keymap.set` is ready to print.

File: lua/nvim-tree/legacy.lua (hunk 2 above)
Comment (kyazdani42) [reply lua/nvim-tree/legacy.lua:301]: Yes, also i've noticed from some PRs on nvim-treesitter that vim.keymap.set takes an optional "desc" argument which we should use so the nvim-tree will work in which-key and allow us to use that in the help menu

**Comment by alex-courtis** [comment]
See #1579

**Pull request closed without merging**